src/bin/driver.rs

```diff
@@ -487,6 +505,46 @@ mod app {
 
         let target = new_settings.stream_target.into();
         c.shared.network.lock(|net| net.direct_stream(target));
+
+        (c.shared.output_state, c.shared.relay_state).lock(|output, relay| {
+            for (i, (new_output_enabled, old_output_enabled)) in new_settings
+                .output_enabled
+                .iter()
+                .zip(old_settings.output_enabled)
+                .enumerate()
+            {
+                if *new_output_enabled && !old_output_enabled {
```

Comment:
You can get rid of having to deref here if you do `for (i, (&new_output_enabled, ...` above.